### PR TITLE
Fixes for OS/OG

### PR DESCRIPTION
**Pull request by muratovv**
### Description of the Change
* remove setPcs method from OG
* minor issues with clang-format
* add max transaction size to OG. Not it shares only a limited number of transactions.
* OS queue replaced with an unordered set for checking not-unique transactions

### Benefits

Performance improvements and DDoS prevention

### Possible Drawbacks 
The absence of tests. I think we have to discuss it during the review

### CODE REVIEW

File: irohad/ordering/impl/on_demand_ordering_service_impl.cpp [hunk 1]
```diff
@@ -118,34 +118,26 @@ OnDemandOrderingServiceImpl::onRequestProposal(consensus::Round round) {
...
     collection.insert(
         std::end(collection),
-        std::make_move_iterator(std::begin(batch->transactions())),
-        std::make_move_iterator(std::end(batch->transactions())));
-  }
-
-  if (discarded_txs_amount) {
-    *discarded_txs_amount = 0;
-    while (tx_batches_queue.try_pop(batch)) {
-      *discarded_txs_amount += boost::size(batch->transactions());
```
Comment (MBoldyrev) [inline irohad/ordering/impl/on_demand_ordering_service_impl.cpp]: why did you remove this? now the `discarded_txs_amount` it not set.

File: irohad/ordering/impl/on_demand_ordering_gate.cpp [hunk 2]
```diff
@@ -120,6 +101,35 @@ OnDemandOrderingGate::processProposalRequest(
...
+                   // no blocks committed, no transactions to remove
+                 });
+
+  auto batches = cache_->pop();
+  cache_->addToBack(batches);
+
+  // get only transactions which fit to next proposal
+  auto end_iterator = batches.begin();
+  if (batches.size() > transaction_limit_) {
+    std::advance(end_iterator, transaction_limit_);
```
Comment (MBoldyrev) [inline irohad/ordering/impl/on_demand_ordering_gate.cpp]: You count batches as if they were transactions. Is it the intended behaviour?

File: irohad/ordering/impl/on_demand_ordering_gate.hpp [hunk 3]
```diff
@@ -86,13 +84,17 @@ namespace iroha {
               std::shared_ptr<const OnDemandOrderingService::ProposalType>>
               proposal) const;
 
+      void sendTransactionsOS(const BlockRoundEventType &event);
```
Comment (lebdron) [inline irohad/ordering/impl/on_demand_ordering_gate.hpp]:
```suggestion
      void sendCachedTransactions(const BlockRoundEventType &event);
```

File: irohad/main/impl/on_demand_ordering_init.hpp [hunk 4]
```diff
@@ -75,6 +75,7 @@ namespace iroha {
           consensus::Round initial_round,
           std::function<std::chrono::milliseconds(
               const synchronizer::SynchronizationEvent &)> delay_func,
+          size_t max_number_of_transactions,
```
Comment (lebdron) [inline irohad/main/impl/on_demand_ordering_init.hpp]: Please make this variable name consistent with `max_size` in `initOrderingGate`, maybe by renaming `max_size` to `max_number_of_transactions`.

File: irohad/ordering/impl/on_demand_ordering_service_impl.cpp (hunk 1 above)
Comment (lebdron) [reply irohad/ordering/impl/on_demand_ordering_service_impl.cpp]: It was expected to be replaced with atomic counter in `onBatches` method, so that there are no transactions to discard. However, since it is not introduced yet, I would suggest to make the following changes:
```C++
  auto it = tx_batches_queue.begin();
  for (;
       it != tx_batches_queue.end() and collection.size() < requested_tx_amount;
       ++it) {
    collection.insert(std::end(collection),
                      std::begin((*it)->transactions()),
                      std::end((*it)->transactions()));
  }

  if (discarded_txs_amount) {
    *discarded_txs_amount = 0;
    for (; it != tx_batches_queue.end(); ++it) {
      *discarded_txs_amount += boost::size((*it)->transactions());
    }
  }
```

File: libs/common/CMakeLists.txt [hunk 5]
```diff
@@ -47,7 +47,7 @@ if (EXISTS "${PROJECT_SOURCE_DIR}/.git")
     # Get pretty version
     execute_process(COMMAND
         "${GIT_EXECUTABLE}"
-        describe --always
+        describe --tags
```
Comment (MBoldyrev) [inline libs/common/CMakeLists.txt]:
```suggestion
        describe --tags --always
```

File: irohad/ordering/impl/on_demand_ordering_service_impl.cpp (hunk 1 above)
Comment (MBoldyrev) [reply irohad/ordering/impl/on_demand_ordering_service_impl.cpp]: whoops. please either remove the discarded txs counter at all, or set it in `getTransactions`
